### PR TITLE
ZER-26  Setup Rubocop per-commit hook

### PR DESCRIPTION
**Pull request by aboriiisova**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: .pronto.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  exclude:
+    - 'app/assets/**/*'
+github:
+  slug: prontolabs/pronto
+  access_token: B26354
+  api_endpoint: https://api.github.com/
+  web_endpoint: https://github.com/
+max_warnings: 150
+warnings_per_review: 30
+verbose: false
```
Comment (maksymst) [inline .pronto.yml:15]: Add a new line at the end of the file. Github displays an icon at the last line to highlight the issue.
Read more info in https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline
It is possible that you can configure your editor to add the empty line automatically. Check the editor setting for more info.

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  
+  Exclude:
+    - 'vendor/**/*'
+    - 'tmp/**/*'
+    - 'lib/**/*'
+    - 'test/**/*'
+    - 'bin/*'
+    - 'config/**/*.yml'
+    - 'db/**/*'
+    - 'db/schema.rb'
```
Comment (maksymst) [inline .rubocop.yml:26]: new line at EOF. see above

**Review by maksymst — CHANGES_REQUESTED** [review]

File: .pronto.yml (hunk 1 above)
Comment (aboriiisova) [reply .pronto.yml]: Fixed in last commit

File: README.md [hunk 3]
```diff
@@ -104,6 +104,33 @@ $ brew install SOMEREPOproductions
...
+$ ```rubocop app spec lib/something.rb```
+For more details check the available command-line options:
+
+$ ```rubocop -h```
+
+Create a new file named ```pre-commit``` inside ```.git/hooks``` directory and paste the following content:
+
+#.git/hooks/pre-commit
+#!/bin/sh
+```bundle exec pronto run --exit-code```
```
Comment (maksymst) [inline README.md:124]: Remove this

File: README.md [hunk 4]
```diff
@@ -104,6 +104,33 @@ $ brew install SOMEREPOproductions
...
+
+Make the file executable with ```chmod +x .git/hooks/pre-commit```
```
Comment (maksymst) [inline README.md:126]: Run this locally, remove this line and commit if file is changed

File: README.md [hunk 5]
```diff
@@ -104,6 +104,33 @@ $ brew install SOMEREPOproductions
...
+
+For using ```git-hook-pre-commit``` type ```cp git-hooks/pre-commit .git/hooks/pre-commit``` command to install your hook.
+
+Run ```git commit -m "name"``` to commit changes locally.
+If you have some troubles with style conventions after running ```git commit -m "name"```, you need to run ```rubocop -a``` or ```rubocop -A```. Each of these commands cans resolve the majority of warnings.
```
Comment (maksymst) [inline README.md:131]:
```suggestion
If you have some troubles with style conventions after running ```git commit -m "name"```, you need to run ```rubocop -a``` or ```rubocop -A```. Each of these commands can resolve the majority of warnings.
```

**Review by maksymst — CHANGES_REQUESTED** [review]

**Pull request closed without merging**